remote/mdb_util_test.go

```diff
@@ -12,30 +13,101 @@ import (
 )
 
 func makeTestMDBServiceAndClient(ctx context.Context, mngr jasper.Manager) (Manager, error) {
-	addr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("localhost:%d", testutil.GetPortNumber()))
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
+	var (
+		addr   net.Addr
+		client Manager
+		err    error
+	)
+tryPort:
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, errors.Wrap(ctx.Err(), "starting MDB service")
+		default:
+			addr, err = net.ResolveTCPAddr("tcp", fmt.Sprintf("localhost:%d", testutil.GetPortNumber()))
+			if err != nil {
+				continue tryPort
+			}
 
-	closeService, err := StartMDBService(ctx, mngr, addr)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-	go func() {
-		<-ctx.Done()
-		grip.Notice(closeService())
-	}()
-	if err = testutil.WaitForWireService(ctx, addr); err != nil {
-		return nil, errors.WithStack(err)
-	}
+			srvCtx, srvCancel := context.WithCancel(ctx)
+			closeService, err := StartMDBService(srvCtx, mngr, addr)
+			if err != nil {
+				srvCancel()
+				return nil, errors.WithStack(err)
+			}
+
+			failedToConnect := make(chan struct{})
+			go func() {
+				defer func() {
+					srvCancel()
+					grip.Notice(closeService())
+				}()
+				select {
+				case <-ctx.Done():
+				case <-failedToConnect:
+				}
+			}()
 
-	client, err := NewMDBClient(ctx, addr, 0)
-	if err != nil {
-		return nil, errors.WithStack(err)
+			client, err = tryConnectToMDBService(ctx, addr)
+			if err != nil {
+				close(failedToConnect)
```

Comment:
if we can't connect (for some reason) will we loop infinitely?

Reply:
No because there's a context done check at the top of the infinite for loop (which is the only case in which we would stop trying to initialize the server/client).